backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/businessentities/comparators/VmsCpuPinningPolicyComparator.java

```diff
@@ -0,0 +1,28 @@
+package org.ovirt.engine.core.common.businessentities.comparators;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+import org.ovirt.engine.core.common.businessentities.VM;
+
+/**
+ * Comparing VMs based on the CPU pinning policy and if needed, their CPU topology.
+ */
+public class VmsCpuPinningPolicyComparator implements Comparator<VM>, Serializable {
+
+    @Override
+    public int compare(VM o1, VM o2) {
+        // Check the policy, their values are from the higher (more restricted policy) to the lower.
+        if (o2.getCpuPinningPolicy().getValue() - o1.getCpuPinningPolicy().getValue() != 0) {
+            return o2.getCpuPinningPolicy().getValue() - o1.getCpuPinningPolicy().getValue();
+        }
+        // The policies are equal, compare based on CPU topology.
+        if (o2.getNumOfSockets() - o1.getNumOfSockets() != 0) {
+            return o2.getNumOfSockets() - o1.getNumOfSockets();
```

Comment:
it would be better to not repeat the computation twice every time

about the logic here - wouldn't the amount of CPUs be more significant than the topology itself?

Reply:
I do not think so, I think that placing a VM 8-1-1 is much easier than 1-1-8.